### ISSUE TITLE
`Azure Support` fails v1.5.10 spring-boot project

### ISSUE THREAD

**Issue by Eskibear** [Microsoft/azure-spring-boot#307]
## Environment
- Spring boot starter:
  - `Azure Support` in **Spring Initializr**
  
- OS Type: MacOS (I suppose the issue is cross-platform)
- Java version: 
  - 1.8
  

## Summary
Add `Azure Support` as a dependency, the spring-boot project fails to build.

## Reproduce steps
* open https://start.spring.io/
* select spring-boot version `1.5.10` 
* search and add `Azure Support` as the only dependency
* click `Generate Project` to download the zip file, and unzip it.
* in the folder, `mvn package` fails.
* even though `mvn package -DskipTests` can do the packaging, `mvn spring-boot:run` still fails with error.
 

## Expected Results
The project successfully runs.

## Actual Results
```
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.maven.AbstractRunMojo$LaunchRunner.run(AbstractRunMojo.java:527)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoClassDefFoundError: org/apache/commons/logging/LogFactory
	at org.springframework.boot.SpringApplication.<clinit>(SpringApplication.java:179)
	at com.example.demo.DemoApplication.main(DemoApplication.java:10)
	... 6 more
Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 8 more
```

**Comment by sophiaso** [comment]
@Eskibear Thanks for reporting the issue, we have fixed the issue but not release yet. Please add the missing dependencies in above PR to your project for temp solution. If you are going to use Spring Boot Starter, then above missing dependencies are not necessary.

**Comment by sophiaso** [comment]
2.0.x released, refs #323 

0.1.x not released

**Comment by sophiaso** [comment]
@Eskibear Thanks and just released a new version 0.2.4 for Spring Boot 1.x.x. 

PR to initializr. https://github.com/spring-io/initializr/pull/691

**Comment by sophiaso** [comment]
Generate project on initializr by selecting `Azure Support`, compilation no longer fails.
